drivers/rtc/rtc_smartbond.c

```diff
@@ -0,0 +1,632 @@
+
+	bcd_to_alarm_calendar(timeptr);
+	bcd_to_alarm_time(timeptr);
+	*mask = rtc_to_tm_alarm_mask(RTC->RTC_ALARM_ENABLE_REG);
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+
+static int rtc_smartbond_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
+
+	unsigned int key;
+	int status;
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	/* Globally disable interrupts as the status flag can be updated within ISR */
+	key = DA1469X_IRQ_DISABLE();
+	status = data->is_alarm_pending;
+	/* After reading, the alarm status should be cleared. */
+	data->is_alarm_pending = 0;
+	DA1469X_IRQ_ENABLE(key);
+
+	return status;
+}
+
+static int rtc_smartbond_alarm_set_callback(const struct device *dev, uint16_t id,
+		rtc_alarm_callback callback, void *user_data)
+{
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	data->alarm_cb = callback;
+	data->alarm_user_data = user_data;
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+
+static int rtc_smartbond_alarm_get_supported_fields(const struct device *dev, uint16_t id,
+						uint16_t *mask)
+{
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
+
+	if (mask == NULL) {
+		LOG_ERR("Pointer to store the mask value is missed");
+		return -EINVAL;
+	}
+
+	*mask = (uint16_t)RTC_SMARTBOND_SUPPORTED_ALARM_FIELDS;
+
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_RTC_UPDATE)
+static int rtc_smartbond_update_set_callback(const struct device *dev, rtc_update_callback callback,
+						void *user_data)
+{
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	data->update_cb = callback;
+	data->update_user_data = user_data;
+
+	if (data->update_cb) {
+		/* Enable asserting the RTC interrupt line when the second counter rolls over. */
+		RTC->RTC_INTERRUPT_ENABLE_REG = RTC_RTC_INTERRUPT_ENABLE_REG_RTC_SEC_INT_EN_Msk;
+	} else {
+		RTC->RTC_INTERRUPT_DISABLE_REG = RTC_RTC_INTERRUPT_DISABLE_REG_RTC_SEC_INT_DIS_Msk;
+	}
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+#endif
+
+struct rtc_driver_api rtc_smartbond_driver_api = {
+	.get_time = rtc_smartbond_get_time,
+	.set_time = rtc_smartbond_set_time,
+#if defined(CONFIG_RTC_ALARM)
+	.alarm_get_time = rtc_smartbond_alarm_get_time,
+	.alarm_set_time = rtc_smartbond_alarm_set_time,
+	.alarm_is_pending = rtc_smartbond_alarm_is_pending,
+	.alarm_set_callback = rtc_smartbond_alarm_set_callback,
+	.alarm_get_supported_fields = rtc_smartbond_alarm_get_supported_fields,
+#endif
+#if defined(CONFIG_RTC_UPDATE)
+	.update_set_callback = rtc_smartbond_update_set_callback,
+#endif
+#if defined(CONFIG_RTC_CALIBRATION)
+	.set_calibration = NULL,
+	.get_calibration = NULL,
+#endif
```

Comment:
No need for this block, this struct is initialized at build time so any unset field would be zeroed out automatically

Reply:
Fixed